src/pip/_internal/req/req_set.py

```diff
@@ -167,6 +166,10 @@ def add_requirement(
         # If we're now installing a constraint, mark the existing
         # object for real installation.
         existing_req.constraint = False
+        # If we're now installing a top level requirement, mark the existing
+        # object as top level.
```

Comment:
Maybe this should say "user supplied" rather than "top level" - why introduce yet another term for this idea just for this one comment?

Reply:
Yes, good catch. I adjusted the vocabulary also in the news fragment.